test/token/ERC20/ERC20.behavior.js

```diff
@@ -1,335 +1,270 @@
-    });
+  it('total supply: returns the total token value', async function () {
+    expect(await this.token.totalSupply()).to.equal(initialSupply);
   });
 
   describe('balanceOf', function () {
-    describe('when the requested account has no tokens', function () {
-      it('returns zero', async function () {
-        expect(await this.token.balanceOf(anotherAccount)).to.be.bignumber.equal('0');
-      });
+    it('returns zero when the requested account has no tokens', async function () {
+      expect(await this.token.balanceOf(this.anotherAccount)).to.equal(0n);
     });
 
-    describe('when the requested account has some tokens', function () {
-      it('returns the total token value', async function () {
-        expect(await this.token.balanceOf(initialHolder)).to.be.bignumber.equal(initialSupply);
-      });
+    it('returns the total token value when the requested account has some tokens', async function () {
+      expect(await this.token.balanceOf(this.initialHolder)).to.equal(initialSupply);
     });
   });
 
   describe('transfer', function () {
-    shouldBehaveLikeERC20Transfer(initialHolder, recipient, initialSupply, function (from, to, value) {
-      return this.token.transfer(to, value, { from });
+    beforeEach(function () {
+      this.transfer = (from, to, value) => this.token.connect(from).transfer(to, value);
     });
+
+    shouldBehaveLikeERC20Transfer(initialSupply);
   });
 
   describe('transfer from', function () {
-    const spender = recipient;
-
     describe('when the token owner is not the zero address', function () {
-      const tokenOwner = initialHolder;
-
       describe('when the recipient is not the zero address', function () {
-        const to = anotherAccount;
-
         describe('when the spender has enough allowance', function () {
           beforeEach(async function () {
-            await this.token.approve(spender, initialSupply, { from: initialHolder });
+            await this.token.connect(this.initialHolder).approve(this.recipient, initialSupply);
           });
 
           describe('when the token owner has enough balance', function () {
             const value = initialSupply;
 
-            it('transfers the requested value', async function () {
-              await this.token.transferFrom(tokenOwner, to, value, { from: spender });
-
-              expect(await this.token.balanceOf(tokenOwner)).to.be.bignumber.equal('0');
+            beforeEach(async function () {
+              this.tx = await this.token
+                .connect(this.recipient)
+                .transferFrom(this.initialHolder, this.anotherAccount, value);
+            });
 
-              expect(await this.token.balanceOf(to)).to.be.bignumber.equal(value);
+            it('transfers the requested value', async function () {
+              expect(await this.token.balanceOf(this.initialHolder)).to.equal(0n);
+              expect(await this.token.balanceOf(this.anotherAccount)).to.equal(value);
```

Comment:
await expect(this.tx)..to.changeTokenBalances(this.token, [ this.initialHolder, this.anotherAccount], [ -value, value ]);